### PR TITLE
Duplicated declaration does not lead to wrong analysis result

### PR DESCRIPTION
**Pull request by jjohannes**
If there are two declarations to the same component with different scopes, e.g.

```
dependencies {
  runtimeOnly("org.apache.commons:commons-collections4:4.4")
  implementation("org.apache.commons:commons-collections4:4.4")
}
```

the analysis discarded one of them by using 'find' instead of 'filter' in StandardTransform.kt. This can lead to a wrong recommendation. In the example provided in the test, the plugin recommended to remove the 'implementation' dependency.

A related feature could be that the plugin also detects the duplication, and recommends to remove the unneeded declaration - in this case `runtimeOnly("...")`. Doing this, would be an additional analysis imo, as the plugin would need to check how the different configurations relate. And maybe this is also not wanted in all cases. The setup is not "wrong" as the classpaths don't change by removing the duplicates. It may be out of scope of this plugin altogether.

This fix is only about not reporting something wrong.

### CODE REVIEW

**Comment by autonomousapps** [comment]
> A related feature could be that the plugin also detects the duplication, and recommends to remove the unneeded declaration

The plugin actually used to do this, pre-1.0. I forget all the context, but I stopped doing it after some complaints surfaced about false positives.

File: src/main/kotlin/com/autonomousapps/transform/StandardTransform.kt [hunk 1]
```diff
@@ -103,38 +103,40 @@ internal class StandardTransform(
     val usageIter = usages.iterator()
     while (usageIter.hasNext()) {
       val usage = usageIter.next()
-      val decl = declarations.find { it.variant(supportedSourceSets) == usage.variant }
+      val declarationsForVariant = declarations.filter { it.variant(supportedSourceSets) == usage.variant }.toSet()
```
Comment (autonomousapps) [inline src/main/kotlin/com/autonomousapps/transform/StandardTransform.kt:106]: there's a `filterToSet` extension function that I'd prefer to use to reduce intermediate object creation.

**Review by autonomousapps — CHANGES_REQUESTED** [review]
Thanks for the contribution! I have one minor change request.

File: src/main/kotlin/com/autonomousapps/transform/StandardTransform.kt (hunk 1 above)
Comment (jjohannes) [reply src/main/kotlin/com/autonomousapps/transform/StandardTransform.kt]: Done.

**Review by autonomousapps — APPROVED** [review]

**Pull request merged**